src/atlas/interpolation/method/bil/BilinearRemapping.cc

```diff
@@ -0,0 +1,395 @@
+            //while (lon < 0.0) { lon += 180.0; }
+            PointXY p{lon, (*olonlat_)(ip, 1)};  // lookup point
+
+            idx_t kpts   = 1;
+            bool success = false;
+            std::ostringstream failures_log;
+
+            while (!success && kpts <= maxNbElemsToTry) {
+                max_neighbours = std::max(kpts, max_neighbours);
+
+                ElemIndex2::NodeList cs = eTree->kNearestNeighbours(p, kpts);
+                Triplets triplets       = projectPointToElements(ip, cs, failures_log);
+
+                if (triplets.size()) {
+                    std::copy(triplets.begin(), triplets.end(), std::back_inserter(weights_triplets));
+                    success = true;
+                }
+                kpts *= 2;
+            }
+
+            if (!success) {
+                failures.push_back(ip);
+                if (not treat_failure_as_missing_value_) {
+                    Log::debug() << "------------------------------------------------------"
+                                    "---------------------\n";
+                    Log::debug() << "Failed to project point (lon,lat)=" << p << '\n';
+                    Log::debug() << failures_log.str();
+                }
+            }
+        }
+    }
+    Log::debug() << "Maximum neighbours searched was " << eckit::Plural(max_neighbours, "element") << std::endl;
+
+    if (failures.size()) {
+        if (treat_failure_as_missing_value_) {
+            missing_.resize(failures.size());
+            std::copy(std::begin(failures), std::end(failures), missing_.begin());
+        }
+        else {
+            // If this fails, consider lowering atlas::grid::parametricEpsilon
+            std::ostringstream msg;
+            msg << "Rank " << mpi::rank() << " failed to project points:\n";
+            for (std::vector<size_t>::const_iterator i = failures.begin(); i != failures.end(); ++i) {
+                const PointLonLat pll{(*olonlat_)(*i, (size_t)0), (*olonlat_)(*i, (size_t)1)};  // lookup point
+                msg << "\t(lon,lat) = " << pll << "\n";
+            }
+
+            Log::error() << msg.str() << std::endl;
+            throw_Exception(msg.str());
+        }
+    }
+
+    // fill sparse matrix and return
+    Matrix A(out_npts, inp_npts, weights_triplets);
+    matrix_shared_->swap(A);
+}
+
+Method::Triplets BilinearRemapping::projectPointToElements(size_t ip, const ElemIndex2::NodeList& elems,
+                                                           std::ostream& /* failures_log */) const {
+    ATLAS_ASSERT(elems.begin() != elems.end());
+
+    const size_t inp_points = ilonlat_->shape(0);
+    std::array<size_t, 4> idx;
+    std::array<double, 4> w;
+
+    Triplets triplets;
+    triplets.reserve(4);
+
+    double lon{(*olonlat_)(ip, 0)};
+    while (lon > 360.0) { lon -= 360.0; }
+    while (lon < 0.0) { lon += 360.0; }
+    PointXY ob_loc{lon, (*olonlat_)(ip, 1)};  // lookup point
+
+    idx_t single_point;
```

Comment:
Unused variable

Reply:
`single_point` removed in latest commit.